homeassistant/components/zha/device_tracker.py

```diff
@@ -0,0 +1,115 @@
+"""Support for the ZHA platform."""
+import logging
+import time
+from homeassistant.components.device_tracker import (
+    SOURCE_TYPE_ZIGBEE, DOMAIN
+)
+from homeassistant.components.device_tracker.config_entry import (
+    DeviceTrackerEntity
+)
+from homeassistant.const import (
+    STATE_NOT_HOME,
+    STATE_HOME
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from .core.const import (
+    DATA_ZHA, DATA_ZHA_DISPATCHERS, ZHA_DISCOVERY_NEW,
+    POWER_CONFIGURATION_CHANNEL, SIGNAL_STATE_ATTR,
+    SIGNAL_ATTR_UPDATED
+)
+from .entity import ZhaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Zigbee Home Automation device tracker from config entry."""
+    async def async_discover(discovery_info):
+        await _async_setup_entities(hass, config_entry, async_add_entities,
+                                    [discovery_info])
+
+    unsub = async_dispatcher_connect(
+        hass, ZHA_DISCOVERY_NEW.format(DOMAIN), async_discover)
+    hass.data[DATA_ZHA][DATA_ZHA_DISPATCHERS].append(unsub)
+
+    device_trackers = hass.data.get(DATA_ZHA, {}).get(DOMAIN)
+    if device_trackers is not None:
+        await _async_setup_entities(hass, config_entry, async_add_entities,
+                                    device_trackers.values())
+        del hass.data[DATA_ZHA][DOMAIN]
+
+    return True
+
+
+async def _async_setup_entities(hass, config_entry, async_add_entities,
+                                discovery_infos):
+    """Set up the ZHA device trackers."""
+    entities = []
+    for discovery_info in discovery_infos:
+        entities.append(ZHADeviceTrackerEntity(**discovery_info))
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class ZHADeviceTrackerEntity(DeviceTrackerEntity, ZhaEntity):
```

Comment:
This is the wrong approach.

For the device tracker, we need to create multiple base entities. One for apps that push data, one for scanner based approaches. @Kane610 is actually working on an entity to be used in his PR over at #24367

Reply:
That being said, this class could be a good foundation to be used as a base class

Reply:
I'll keep an eye on that and update accordingly when the time comes

Reply:
any pointers on how you want this to work? Maybe I'll be able to finish this before the other PR is ready?

Reply:
Well like we have an entity base class for other integrations, we would need one for device trackers for entities that are "seen" .